### PR TITLE
Patch reward and withdraw scens; and take it easier on sourcing WETH

### PR DESCRIPTION
**Pull request by jflatow**

### CODE REVIEW

File: scenario/context/CometContext.ts [hunk 1]
```diff
@@ -316,18 +320,20 @@ async function getActors(context: CometContext): Promise<{ [name: string]: Comet
 async function getAssets(context: CometContext): Promise<{ [symbol: string]: CometAsset }> {
   const { deploymentManager } = context.world;
 
-  let comet = await context.getComet();
-  let signer = await deploymentManager.getSigner();
-  let numAssets = await comet.numAssets();
-  let assetAddresses = [
+  const COMP =await context.getComp();
```
Comment (kevincheng96) [inline scenario/context/CometContext.ts:323]: nit: missing space

**Review by kevincheng96 — APPROVED** [review]
LGTM

File: scenario/context/CometContext.ts (hunk 1 above)
Comment (jflatow) [reply scenario/context/CometContext.ts]: good catch 🎣 , fixed

**Review by scott-silver — APPROVED** [review]

File: scenario/constraints/NativeTokenConstraint.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+export class NativeTokenConstraint<T extends CometContext, R extends Requirements> implements Constraint<T, R> {
+  async solve(_requirements: R, _context: T) {
+    return [
+      async function (ctx: T): Promise<T> {
+        for (const symbol in ctx.assets) {
+          const contract = await ctx.world.deploymentManager.contract(symbol);
+          if (contract.deposit) {
+            const fauceteer = await ctx.getFauceteer();
+            const amount = 200_000n * 10n**18n;
```
Comment (kevincheng96) [inline scenario/constraints/NativeTokenConstraint.ts:13]: nit: `exp()`?

Comment (jflatow) [reply scenario/constraints/NativeTokenConstraint.ts:13]: Not sure its worth the import?

Comment (jflatow) [reply scenario/constraints/NativeTokenConstraint.ts:13]: But yeah, and we should probably call `contract.decimals()` too

**Pull request merged**